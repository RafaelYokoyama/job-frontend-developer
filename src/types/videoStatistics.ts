import { IconName } from "./Icon";

export interface StatisticsProps  {
    name: IconName;
    value: string;
  };

export interface VideoStatisticsProps  {
  statistics: StatisticsProps[];
};
