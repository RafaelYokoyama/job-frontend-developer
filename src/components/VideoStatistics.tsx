import React from 'react';
import Icon from './Icon';
import { VideoStatisticsProps } from 'types/videoStatistics';

const VideoStatistics = ({ statistics }: VideoStatisticsProps) => {
  return (
    <div className="mt-4 flex items-center">
      {statistics.map((stat, index) => (
        <p key={index} className="flex items-center pr-4">
          <Icon name={stat.name} className="h-5 w-5 mr-1" />
          {stat.value}
        </p>
      ))}
    </div>
  );
};

export default VideoStatistics;
