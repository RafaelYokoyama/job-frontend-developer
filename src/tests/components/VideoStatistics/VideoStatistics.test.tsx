import React from 'react';
import { render } from '@testing-library/react';

import VideoStatistics from 'components/VideoStatistics';
import { mockVideoStatistics } from 'tests/mocks/videoStatistics';

describe('VideoStatistics Component', () => {
  it('checks a snapshot', () => {
    const { container } = render(
      <VideoStatistics statistics={mockVideoStatistics} />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders statistics values correctly', () => {
    const { getByText } = render(
      <VideoStatistics statistics={mockVideoStatistics} />
    );
    mockVideoStatistics.forEach((stat) => {
      const statValueElement = getByText(stat.value);
      expect(statValueElement).toBeInTheDocument();
    });
  });

  it('renders the correct number of statistics elements', () => {
    const { container } = render(
      <VideoStatistics statistics={mockVideoStatistics} />
    );
    const statisticElements = container.querySelectorAll(
      '.flex.items-center.pr-4'
    );
    expect(statisticElements.length).toBe(mockVideoStatistics.length);
  });

  it('renders the statistics in the correct order', () => {
    const { container } = render(
      <VideoStatistics statistics={mockVideoStatistics} />
    );
    const renderedStatistics = Array.from(
      container.querySelectorAll('.flex.items-center.pr-4')
    ).map((element) => element.textContent);
    const expectedStatisticsOrder = mockVideoStatistics.map(
      (stat) => `${stat.value}`
    );
    expect(renderedStatistics).toEqual(expectedStatisticsOrder);
  });

  it('renders the statistics with the correct CSS class', () => {
    const { container } = render(
      <VideoStatistics statistics={mockVideoStatistics} />
    );
    const statisticElements = container.querySelectorAll(
      '.flex.items-center.pr-4'
    );
    statisticElements.forEach((element) => {
      expect(element).toHaveClass('flex items-center pr-4');
    });
  });
});
