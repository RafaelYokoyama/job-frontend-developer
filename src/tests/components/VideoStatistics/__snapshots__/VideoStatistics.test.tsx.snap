// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoStatistics Component checks a snapshot 1`] = `
<div>
  <div
    class="mt-4 flex items-center"
  >
    <p
      class="flex items-center pr-4"
    >
      <svg
        class="h-5 w-5 mr-1"
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 10v12"
        />
        <path
          d="M15 5.88 14 10h5.83a2 2 0 0 1 1.92 2.56l-2.33 8A2 2 0 0 1 17.5 22H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h2.76a2 2 0 0 0 1.79-1.11L12 2h0a3.13 3.13 0 0 1 3 3.88Z"
        />
      </svg>
      100
    </p>
    <p
      class="flex items-center pr-4"
    >
      <svg
        class="h-5 w-5 mr-1"
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 10v12"
        />
        <path
          d="M15 5.88 14 10h5.83a2 2 0 0 1 1.92 2.56l-2.33 8A2 2 0 0 1 17.5 22H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h2.76a2 2 0 0 0 1.79-1.11L12 2h0a3.13 3.13 0 0 1 3 3.88Z"
        />
      </svg>
      20
    </p>
    <p
      class="flex items-center pr-4"
    >
      <svg
        class="h-5 w-5 mr-1"
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 10v12"
        />
        <path
          d="M15 5.88 14 10h5.83a2 2 0 0 1 1.92 2.56l-2.33 8A2 2 0 0 1 17.5 22H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h2.76a2 2 0 0 0 1.79-1.11L12 2h0a3.13 3.13 0 0 1 3 3.88Z"
        />
      </svg>
      1000
    </p>
  </div>
</div>
`;
